Sources/SwiftKueryPostgreSQL/PostgreSQLConnection.swift

```diff
@@ -356,8 +356,14 @@ public class PostgreSQLConnection: Connection {
         }
         // Remove entry from the preparedStatements set
         preparedStatements.remove(statement.name)
-        // No need to deallocate prepared statements in PostgreSQL.
-        return runCompletionHandler(.successNoData, onCompletion: onCompletion)
+        // Deallocate prepared statement
+        var result:QueryResult?
+        self.execute("DEALLOCATE \"\(statement.name)\"") { (dealloc_result:QueryResult) in
+            if let _ = dealloc_result.asError {
+                result = dealloc_result
```

Comment:
The completion handler for execute is called asynchronously - I think the below call to `runCompletionHandler` needs to be nested inside this handler.

Reply:
So either scrap this pull request or edit it to use the new async approach, but the bottom line is the important thing is to use to call `"DEALLOCATE \"\(statement.name)\""`, otherwise you may end up with serious issues in the server - in my case, the database server just died every couple of hours because it ran out of memory.